development/build/scripts.js

```diff
@@ -205,171 +216,250 @@ function createScriptTasks({ browserPlatforms, livereload }) {
-        METAMASK_ENVIRONMENT: environment,
-        METAMASK_VERSION: baseManifest.version,
-        NODE_ENV: opts.devMode ? 'development' : 'production',
-        IN_TEST: opts.testing ? 'true' : false,
-        PUBNUB_SUB_KEY: process.env.PUBNUB_SUB_KEY || '',
-        PUBNUB_PUB_KEY: process.env.PUBNUB_PUB_KEY || '',
-        CONF: opts.devMode ? conf : {},
-        SENTRY_DSN: process.env.SENTRY_DSN,
-        INFURA_PROJECT_ID: opts.testing
-          ? '00000000000000000000000000000000'
-          : conf.INFURA_PROJECT_ID,
-        SEGMENT_HOST: conf.SEGMENT_HOST,
-        // When we're in the 'production' environment we will use a specific key only set in CI
-        // Otherwise we'll use the key from .metamaskrc or from the environment variable. If
-        // the value of SEGMENT_WRITE_KEY that we envify is undefined then no events will be tracked
-        // in the build. This is intentional so that developers can contribute to MetaMask without
-        // inflating event volume.
-        SEGMENT_WRITE_KEY:
-          environment === 'production'
-            ? process.env.SEGMENT_PROD_WRITE_KEY
-            : conf.SEGMENT_WRITE_KEY,
-        SEGMENT_LEGACY_WRITE_KEY:
-          environment === 'production'
-            ? process.env.SEGMENT_PROD_LEGACY_WRITE_KEY
-            : conf.SEGMENT_LEGACY_WRITE_KEY,
+function setupMinification(buildConfiguration) {
+  const { events } = buildConfiguration;
+  events.on('configurePipeline', ({ pipeline }) => {
+    pipeline.get('minify').push(
+      terser({
+        mangle: {
+          reserved: ['MetamaskInpageProvider'],
+        },
+        sourceMap: {
+          content: true,
+        },
       }),
-      {
-        global: true,
-      },
     );
+  });
+}
 
-    // Live reload - minimal rebundle on change
-    if (opts.devMode) {
-      bundler = watchify(bundler);
-      // on any file update, re-runs the bundler
-      bundler.on('update', () => {
-        performBundle();
-      });
-    }
+function setupSourcemaps(buildConfiguration, { devMode }) {
+  const { events } = buildConfiguration;
+  events.on('configurePipeline', ({ pipeline }) => {
+    pipeline.get('sourcemaps:init').push(sourcemaps.init({ loadMaps: true }));
+    pipeline
+      .get('sourcemaps:write')
+      // Use inline source maps for development due to Chrome DevTools bug
+      // https://bugs.chromium.org/p/chromium/issues/detail?id=931675
+      .push(
+        devMode
+          ? sourcemaps.write()
+          : sourcemaps.write('../sourcemaps', { addComment: false }),
+      );
+  });
+}
 
-    return bundler;
+async function bundleIt(buildConfiguration) {
+  const { bundlerOpts, events } = buildConfiguration;
+  const bundler = browserify(bundlerOpts);
+  // manually apply non-standard option
+  bundler.external(bundlerOpts.manualExternal);
+  // output build logs to terminal
+  bundler.on('log', log);
+  // forward update event (used by watchify)
+  bundler.on('update', () => performBundle());
```

Comment:
This event used to be setup only in dev mode :thinking: Is it possible this could get triggered during a non-dev build, resulting in `performBundle` being called multiple times?

Reply:
I did attempt to make changes during a `yarn dist` build but was unsuccessful in triggering this.

Reply:
`update` is only emitted by watchify